src/test/java/org/elasticsearch/test/store/MockFSDirectoryService.java

```diff
@@ -89,7 +89,7 @@ public void beforeIndexShardClosed(ShardId sid, @Nullable IndexShard indexShard,
                             // so that even in tests where don't flush we can check the integrity of the Lucene index
                             if (indexShard.engine().hasUncommittedChanges()) { // only if we have any changes
                                 logger.info("{} flushing in order to run checkindex", indexShard.shardId());
-                                Releasables.close(indexShard.engine().snapshotIndex()); // Keep translog for tests that rely on replaying it
+                                Releasables.close(indexShard.engine().snapshotIndex(true)); // Keep translog for tests that rely on replaying it
```

Comment:
it's a non translog trimming commit. The new behavior changes behavior a bit that *if* someone calls new translog out of the blue we will commit it to lucene, but that doesn't happen. I'm not sure why this is needed, it was there before...